cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AppFabricServiceManager.java

```diff
@@ -128,4 +144,63 @@ public void restartAllInstances() {
   public void restartInstances(int instanceId, int... moreInstanceIds) {
     // no-op
   }
+
+  @Override
+  public void updateServiceLogLevels(Map<String, LogEntry.Level> logLevels) {
+    LoggerContext context = getLoggerContext();
+    if (context != null) {
+      for (Map.Entry<String, LogEntry.Level> entry : logLevels.entrySet()) {
+        String loggerName = entry.getKey();
+        LogEntry.Level oldLogLevel = setLogLevel(context, loggerName, entry.getValue());
+        // if logger name is not in oldLogLevels, we need to record its original log level
+        if (!oldLogLevels.containsKey(loggerName)) {
+          oldLogLevels.put(loggerName, oldLogLevel);
+        }
+      }
+    }
+  }
+
+  @Override
+  public void resetServiceLogLevels(Set<String> loggerNames) {
```

Comment:
This is a method which is overridden from MasterServiceManager, the javadoc can be found there.